tests/components/light/test_init.py

```diff
@@ -2,495 +2,623 @@
 
-        common.turn_off(
-            self.hass, entity_id="entity_id_val", transition="transition_val"
-        )
+async def test_methods(hass):
+    """Test if methods call the services as expected."""
+    # Test is_on
+    hass.states.async_set("light.test", STATE_ON)
+    assert light.is_on(hass, "light.test")
 
-        self.hass.block_till_done()
+    hass.states.async_set("light.test", STATE_OFF)
+    assert not light.is_on(hass, "light.test")
 
-        assert 1 == len(turn_off_calls)
-        call = turn_off_calls[-1]
+    # Test turn_on
+    turn_on_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TURN_ON)
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TURN_OFF == call.service
-        assert "entity_id_val" == call.data[ATTR_ENTITY_ID]
-        assert "transition_val" == call.data[light.ATTR_TRANSITION]
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: "entity_id_val",
+            light.ATTR_TRANSITION: "transition_val",
+            light.ATTR_BRIGHTNESS: "brightness_val",
+            light.ATTR_RGB_COLOR: "rgb_color_val",
+            light.ATTR_XY_COLOR: "xy_color_val",
+            light.ATTR_PROFILE: "profile_val",
+            light.ATTR_COLOR_NAME: "color_name_val",
+            light.ATTR_WHITE_VALUE: "white_val",
+        },
+        blocking=True,
+    )
 
-        # Test toggle
-        toggle_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TOGGLE)
+    assert len(turn_on_calls) == 1
+    call = turn_on_calls[-1]
 
-        common.toggle(self.hass, entity_id="entity_id_val", transition="transition_val")
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TURN_ON
+    assert call.data.get(ATTR_ENTITY_ID) == "entity_id_val"
+    assert call.data.get(light.ATTR_TRANSITION) == "transition_val"
+    assert call.data.get(light.ATTR_BRIGHTNESS) == "brightness_val"
+    assert call.data.get(light.ATTR_RGB_COLOR) == "rgb_color_val"
+    assert call.data.get(light.ATTR_XY_COLOR) == "xy_color_val"
+    assert call.data.get(light.ATTR_PROFILE) == "profile_val"
+    assert call.data.get(light.ATTR_COLOR_NAME) == "color_name_val"
+    assert call.data.get(light.ATTR_WHITE_VALUE) == "white_val"
 
-        self.hass.block_till_done()
+    # Test turn_off
+    turn_off_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TURN_OFF)
 
-        assert 1 == len(toggle_calls)
-        call = toggle_calls[-1]
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_OFF,
+        {
+            ATTR_ENTITY_ID: "entity_id_val",
+            light.ATTR_TRANSITION: "transition_val",
+        },
+        blocking=True,
+    )
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TOGGLE == call.service
-        assert "entity_id_val" == call.data[ATTR_ENTITY_ID]
-        assert "transition_val" == call.data[light.ATTR_TRANSITION]
+    assert len(turn_off_calls) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)